### PR TITLE
Introduce CONTRIBUTING.md based on other repositories

### PR DESCRIPTION
**Pull request by B0go**
Also added the commit message as requested here https://github.com/jaegertracing/jaeger/issues/771

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+*Before making any significant changes, please [open an
+issue](https://github.com/jaegertracing/jaeger-kubernetes/issues).* Discussing your proposed
+changes ahead of time will make the contribution process smooth for everyone.
+
+Once we've discussed your changes and you've got your code ready, make sure
+that tests are passing (`make test` or `make cover`) and open your PR. Your
+pull request is most likely to be accepted if it:
+
+* Includes tests for new functionality.
+* Follows the guidelines in [Effective
```
Comment (jpkrohling) [inline CONTRIBUTING.md]: "Effective Go" isn't relevant for this repository

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  review comments](https://github.com/golang/go/wiki/CodeReviewComments).
+* Has a [good commit message](https://chris.beams.io/posts/git-commit/):
+    * Separate subject from body with a blank line
+    * Limit the subject line to 50 characters
+    * Capitalize the subject line
+    * Do not end the subject line with a period
+    * Use the imperative mood in the subject line
+    * Wrap the body at 72 characters
+    * Use the body to explain _what_ and _why_ instead of _how_
+* Each commit must be signed by the author ([see below](#sign-your-work)).
```
Comment (jpkrohling) [inline CONTRIBUTING.md]: Could you add the following items?

* If your change is not storage-dependent, make sure your change is applied consistently across storage templates as well. For instance, changes to the `ConfigMap` might need to be done twice.
* Test your changes and mention in the PR how you tested

**Comment by B0go** [comment]
@jpkrohling done! Please take a look :)

**Comment by B0go** [comment]
@jpkrohling @objectiser forgot to sign the last commits... my bad

Fixed that

**Comment by jpkrohling** [comment]
Thank you for your contribution!

**Pull request merged**